alonzo/formal-spec/utxo.tex

```diff
@@ -601,19 +548,9 @@ \subsection{Witnessing}
     & \where \\
     & ~~~~~~~ \var{txb}~=~\txbody{tx}
     \nextdef
-    & \hspace{-1cm}\fun{checkScriptData} \in \Tx \to \UTxO \to (\ScriptPurpose \times \ScriptHash) \to \Bool \\
-    & \hspace{-1cm}\fun{checkScriptData}~\var{tx}~\var{utxo}~(\var{sp},\var{h})~=~ \exists s, h\mapsto s \in \fun{txscripts}~(\fun{txwits}~tx)\\
-    \land & ~ s \in \ScriptNonNative~\Rightarrow (\fun{indexedRdmrs}~tx~sp \neq \Nothing \\
-    & ~~~~~~ \land \var{sp} \in \UTxOIn \Rightarrow \fun{getData}~tx~utxo~sp \neq \epsilon)
-    \nextdef
-    & \hspace{-1cm}\fun{hashSD} \in \TxWitness \to \ScriptDataHash^? \\
-    & \hspace{-1cm}\fun{hashSD}~\var{txw}~=~
-      \begin{cases}
-        \Nothing & (\var{sc}, \var{dat}, \var{rdmr}) ~=~ (\emptyset, \emptyset, \emptyset) \\
-        \fun{hash}~(\var{sc}, \var{dat}, \var{rdmr}) & \text{otherwise}
-      \end{cases} \\
-    & \hspace{-1cm} ~~\where \\
-    & \hspace{-1cm} ~~~~~~~~~ (\wcard, \var{sc}, \var{dat}, \var{rdmr})\leteq\var{txw}
+    & \hspace{-1cm}\fun{checkRedeemers} \in \Tx \to \UTxO \to (\ScriptPurpose \times \ScriptHash) \to \Bool \\
```

Comment:
script purpose (just the tags) together with the script hash is not enough to get the correct redeemer... for example two UTxO entries can be locked by the same Plutus script but can be getting different redeemers

Reply:
right, I misinterpreted what `ScriptPurpose` is